src/main/java/fr/silverpricing/api/model/Chambre.java

```diff
@@ -13,19 +17,24 @@
 @Setter
 @Getter
 @Entity
-@Table(name = "chambres")
+@Table(name = "chambres",uniqueConstraints = {
+        @UniqueConstraint(columnNames = "id"),
+})
 public class Chambre {
     @Id
-    private String id;
-    @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "residence__id")
-    private Chambre chambre;
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
     private CategoryChambre categoryChambre;
     @OneToMany(mappedBy = "chambre", cascade = CascadeType.ALL, orphanRemoval = true)
     private List<Price> prices = new ArrayList<>();
+    @JsonIgnore
+    @OneToOne(mappedBy = "chambre")
```

Comment:
:warning: The getExternalResidences method contains a complex conditional statement that is hard to read and maintain. (Severity: Major)